src/shell.ts

```diff
@@ -204,7 +205,26 @@ export class Shell {
         else
             e.classList.add("hidden")
     }
-    
+
+    startHourglass(timeout: number) {
```

Comment:
I wonder... Won't it be simpler to add it as part of the watchdog? This way, whenever the watchdog is started the user will get the hourglass and know they're not stuck for ever